src/taipy/core/data/data_node.py

```diff
@@ -285,9 +285,21 @@ def __getattr__(self, attribute_name):
 
     def __get_last_modified_datetime(self) -> Optional[datetime]:
         path = self._properties.get(self.__PATH_KEY, None)
-        if path and os.path.exists(path):
+        if path and os.path.isfile(path):
             return datetime.fromtimestamp(os.path.getmtime(path))
-        return None
+
+        last_modified_datetime = None
+        if path and os.path.isdir(path):
+            for filename in os.listdir(path):
+                filepath = os.path.join(path, filename)
+                if (
+                    os.path.isfile(filepath)
+                    and last_modified_datetime
+                    and datetime.fromtimestamp(os.path.getmtime(filepath)) > last_modified_datetime
+                ):
+                    last_modified_datetime = datetime.fromtimestamp(os.path.getmtime(filepath))
+
+        return last_modified_datetime
```

Comment:
I don't get how it can work. Cause the purpose is to loop over the files and to get the latest time. That makes sense. But we initialize `last_modified_datetime` to `None`, and we update it only if `last_modified_datetime is not None`. So we never update it. Am I missing something?

Reply:
Ah hah, the logic is completely wrong. I wilk make sure we have test case for this.